reconstruction++.cpp

```diff
@@ -1,217 +1,107 @@
-      
-        
+        combine_datapackets_to_fragment(datapackets_clouds,
+                                        interpolated_quaternions,
+                                        data_dir + "/fragments/" + fragment);
     }
 
-
-    std::cout << std::endl << std::endl; 
-
+    std::cout << std::endl << std::endl;
 
     //////////////////////////////////// Odometry ///////////////////////////////////////////////////////////
-
-
-    if (exists(data_dir + "/odometry")) {
-
-
-       std::cout << "Odometry" << std::endl << std::endl; 
-
-
-       std::vector <Dir> odometry; 
-
-
-       path p_odometry(data_dir + "/odometry"); 
-
-       for (auto i = directory_iterator(p_odometry); i != directory_iterator(); ++i) {
-
-
-           std::string odometry_dir = i -> path().filename().string(); 
-
-           if (odometry_dir == ".DS_Store") {continue;} // If Apple
-
-           int no = std::stoi(odometry_dir.substr(odometry_dir.find("_") + 1, odometry_dir.length())); 
-
-           odometry.push_back({odometry_dir, no});  
-
-
-       }
-
-       std::sort(odometry.begin(), odometry.end(), [](Dir i, Dir j) {return i.number < j.number;}); // Sort by number
-
-
-       std::vector <Eigen::Vector3f> translations; 
-
-
-       for (std::size_t i = 0; i < odometry.size(); ++i) {
-
-
-           std::cout << odometry[i].name << std::endl; 
-
-
-           // Read quaternions
-
-           const std::vector <Quaternion_file> quaternions = read_quaternions_file(data_dir + "/odometry/" + odometry[i].name + "/quaternions"); 
-
-
-           // Interpolate quaternions
-
-           const std::vector <Quaternion_file> interpolated_quaternions = interpolate_quaternions(quaternions);
-
-
-           // Load datapackets
-
-           std::vector <std::vector <pcl::PointCloud <pcl::PointXYZ> > > datapacket_clouds = load_datapackets(data_dir + "/odometry/" + odometry[i].name + "/datapackets"); 
-
-
-           // Combine datapackets to scans
- 
-           combine_datapackets_to_scans(datapacket_clouds, interpolated_quaternions, data_dir + "/odometry/" + odometry[i].name); 
-
-
-           // Estimate translation 
-
-           Eigen::Vector3f translation {0, 0, 0}; 
-
-           translation_estimation(data_dir + "/odometry/" + odometry[i].name + "/scans", translation); 
-
-           translations.push_back(translation); 
-
-    
-       }
-
-
-       std::cout << std::endl << std::endl; 
-
-
-       /////////////////////////// Fragment pairwise registration with odometry ////////////////////////////////////////////////////////
-
-
-       std::cout << "Fragment pairwise registration" << std::endl << std::endl; 
-
-
-       std::vector <pcl::PointCloud <pcl::PointXYZ> > fragment_clouds = load_fragments(data_dir + "/fragments"); 
-
-       incremental_pairwise_registration(fragment_clouds, translations, data_dir, visualization); 
-
-
+    if (boost::filesystem::exists(data_dir + "/odometry")) {
```

Comment:
It would have been a bit easier to have the removal of empty lines in a separate PR. But probably too difficult now.

Reply:
Actually, I think to split it up wouldn't be too hard. I'll try and make this into two PRs

Reply:
I'm going to force push this PR without the reformatting commits and make another branch that only applies those

Reply:
Done. This one is now without the reformatting. The build seems to work as well. After we merge this, I'll make a PR with the reformatting, and after that another one where I updated PCL to 1.11 and added the symmetric objective option for ICP.